src/Orleans/Core/ClusterClient.cs

```diff
@@ -143,31 +164,36 @@ public static ClusterClient Create(FileInfo configFile)
         public static ClusterClient Create(ClientConfiguration configuration) => new ClusterClient(configuration);
 
         /// <inheritdoc />
-        public void Start()
+        public async Task Start()
         {
-            lock (this.initLock)
+            this.ThrowIfDisposed();
+            using (await this.initLock.LockAsync())
```

Comment:
probably everything related to start and stop in this class also requires .ConfigureAwait(false), since the most likely scenario is that it will still be using the parent context